components/Form/MultiStepForm.js

```diff
@@ -9,74 +9,63 @@ import Alert from 'components/Alert/Alert';
 import ProgressIndicator from 'components/ProgressIndicator/ProgressIndicator';
 import styles from './MultiStepForm.css';
 
-export class MultiStepForm extends React.Component {
-  static propTypes = {
-    // initialValues must be object where entire form's shape is described
-    initialValues: object.isRequired,
-
-    getErrorMessage: func.isRequired,
-    onEachStepSubmit: func,
-    onFinalSubmit: func.isRequired, // to be considered onSuccess
-    steps: arrayOf(validStep).isRequired,
-  };
-
-  static defaultProps = {
-    onEachStepSubmit: noop,
-  };
-
-  state = {
+MultiStepForm.propTypes = {
+  // initialValues must be object where entire form's shape is described
+  initialValues: object.isRequired,
+
+  getErrorMessage: func.isRequired,
+  onEachStepSubmit: func,
+  onFinalSubmit: func.isRequired, // to be considered onSuccess
+  steps: arrayOf(validStep).isRequired,
+};
+
+MultiStepForm.defaultProps = {
+  onEachStepSubmit: noop,
+};
+
+export function MultiStepForm({
+  steps,
+  initialValues,
+  onEachStepSubmit,
+  onFinalSubmit,
+  getErrorMessage,
+}) {
+  const [state, setState] = useState({
     errorMessage: '',
     stepNumber: 0,
-  };
+  });
 
-  isLastStep = () => {
-    const { steps } = this.props;
-    const { stepNumber } = this.state;
+  const { errorMessage, stepNumber } = state;
 
+  const isLastStep = () => {
     return stepNumber === steps.length - 1;
   };
 
   // We assume this method cannot be called on the last step
-  showNextStep = ({ setFieldTouched }) => {
-    const { steps } = this.props;
-    const { stepNumber } = this.state;
-
+  const showNextStep = ({ setFieldTouched }) => {
     // TODO: Only untouch if value is '' or []
     const nextStepFieldNames = Object.keys(steps[stepNumber + 1].initialValues);
     nextStepFieldNames.forEach(fieldName => setFieldTouched(fieldName, false));
-
-    this.setState(previousState => ({
-      stepNumber: previousState.stepNumber + 1,
-    }));
+    setState({ ...state, stepNumber: state.stepNumber + 1 });
```

Comment:
You'll want to use the implicit callback to avoid race conditions:

```suggestion
    setState(prevState => ({ ...prevState, stepNumber: prevState.stepNumber + 1 }));
```

You'll want to do this on all the state setters in this file, but I like your move to do a state object!

Reply:
done